catalog/spatial/geocoding/spatial-geocoding-offline-catalog/src/main/java/org/codice/ddf/spatial/geocoding/index/GeoNamesCatalogIndexer.java

```diff
@@ -204,25 +192,14 @@ public void updateProgress(final int progress) {
           }
         };
 
-    if (create) {
-      RetryPolicy retryPolicy =
-          new RetryPolicy()
-              .withDelay(10, TimeUnit.SECONDS)
-              .withMaxDuration(5, TimeUnit.MINUTES)
-              .retryOn(Exception.class);
-
-      Failsafe.with(retryPolicy)
-          .run(() -> removeGeoNamesMetacardsFromCatalog(resource, extractionCallback));
-    }
-
     geoEntryExtractor.pushGeoEntriesToExtractionCallback(resource, extractionCallback);
 
-    if (CollectionUtils.isEmpty(metacardList)) {
-      LOGGER.debug("No Metacards were created from the resource.");
+    if (CollectionUtils.isEmpty(geoEntries)) {
+      LOGGER.debug("No metacards were created from the resource.");
       return;
     }
 
-    executeCreateMetacardRequest(metacardList);
+    updateIndex(geoEntries, create, progressCallback, resource);
```

Comment:
So, we automatically update the index when things get added via the GeoNames Catalog Indexer.  How about if things are added other ways?  What if someone used CSW for example to inject a gazetteer record